pkg/rbd/rbd.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright 2024 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package rbd
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/rook/kubectl-rook-ceph/pkg/exec"
+	"github.com/rook/kubectl-rook-ceph/pkg/k8sutil"
+	"github.com/rook/kubectl-rook-ceph/pkg/logging"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// List retrieves and displays the Ceph block pools and their associated images.
+func ListRbdVolumes(ctx context.Context, clientsets *k8sutil.Clientsets, operatorNamespace string, clusterNamespace string) {
+	// List Ceph Block Pools
+	blockPoolList, err := clientsets.Rook.CephV1().CephBlockPools(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("failed to list CephBlockPools: %w", err))
+	}
+
+	// Initialize map to store block pool names and associated Rados namespaces
+	blockPoolNames := make(map[string][]string)
+	for _, blockPool := range blockPoolList.Items {
+		blockPoolNames[blockPool.Name] = []string{"--", "--"}
+	}
+
+	// List Ceph Block Pool Rados Namespaces
+	blockPoolNamespaceList, err := clientsets.Rook.CephV1().CephBlockPoolRadosNamespaces(clusterNamespace).List(ctx, v1.ListOptions{})
+	if err != nil {
+		logging.Fatal(fmt.Errorf("failed to list CephBlockPoolRadosNamespaces: %w", err))
+	}
+	for _, blockPoolNameSpace := range blockPoolNamespaceList.Items {
+		var name string
+		// Check if Spec.BlockPoolName exists; otherwise, use ObjectMeta.Name
+		if blockPoolNameSpace.Spec.BlockPoolName != "" {
+			name = blockPoolNameSpace.Spec.Name
+		} else {
+			name = blockPoolNameSpace.ObjectMeta.Name
+		}
+		if _, exists := blockPoolNames[blockPoolNameSpace.Spec.BlockPoolName]; exists {
+			blockPoolNames[blockPoolNameSpace.Spec.BlockPoolName][1] = name
+		}
+	}
+
+	// Retrieve list of RBD images for each pool
+	cmd := "rbd"
+	for poolName := range blockPoolNames {
+		args := []string{"ls", "--pool=" + poolName}
+		list, err := exec.RunCommandInOperatorPod(ctx, clientsets, cmd, args, operatorNamespace, clusterNamespace, true)
+		if err == nil && len(list) > 0 {
+			blockPoolNames[poolName][0] = strings.ReplaceAll(list, "\n", "")
+		}
```

Comment:
what happens if there is an error?

Reply:
we use the default value `--`   https://github.com/rook/kubectl-rook-ceph/pull/333/files#diff-b4c239a28ac425efccdf14790165fceaf4be580365efb6ccc72068a6800958f8R43

Reply:
This is not the expected behavior, user should be able to distinguish between the error or empty images. please return an error message in case of error

Reply:
What is the expected output? For example, I encounter an error every time I use `builtin-mgr`
```
error(*fmt.wrapError) *{msg: "rbd: error opening pool 'builtin-mgr': (2) No such file or directory\nrbd: listing images failed: (2) No such file or directory\n. failed to run command. command terminated with exit code 2", err: error(*fmt.wrapError) *{msg: "failed to run command. command terminated with exit code 2", err: error(k8s.io/client-go/util/exec.CodeExitError) *(*error)(0xc000414fd0)}}```
![image](https://github.com/user-attachments/assets/3154c5fb-3716-4835-92f7-2fa273da204d)